### PR TITLE
add doc on using compute-sanitizer

### PR DESCRIPTION
**Pull request by rongou**
Signed-off-by: Rong Ou <rong.ou@gmail.com>

### CODE REVIEW

**Review by revans2 — DISMISSED** [review]

File: docs/dev/compute_sanitizer.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+the RAPIDS Spark Plugin is the `memcheck` tool that is capable of precisely detecting and
+attributing out of bounds and misaligned memory access errors in CUDA applications.
+
+To use Compute Sanitizer with the RAPIDS Spark Plugin in a standalone Spark cluster, follow these
+steps:
+  * Create a "fake" java home, for example, in `/opt/compute-sanitizer-java`:
+```bash
+mkdir -p /opt/compute-sanitizer-java/bin
+```
+  * Create a "fake" java binary:
```
[the remark this answers is not in the document]
Comment (rongou) [reply docs/dev/compute_sanitizer.md:34]: Yeah originally I was trying to get this to work with the 2 DGX-2s. Added a single worker mode.

File: docs/dev/compute_sanitizer.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+---
+
+# Using Compute Sanitizer with the RAPIDS Plugin for Spark
+
+Compute Sanitizer is a functional correctness checking suite included in the CUDA toolkit.
+This suite contains multiple tools that can perform different type of checks. Of main interest to
+the RAPIDS Spark Plugin is the `memcheck` tool that is capable of precisely detecting and
+attributing out of bounds and misaligned memory access errors in CUDA applications.
+
+To use Compute Sanitizer with the RAPIDS Spark Plugin in a standalone Spark cluster, follow these
```
[the remark this answers is not in the document]
Comment (rongou) [reply docs/dev/compute_sanitizer.md]: Added a single worker mode.

File: docs/dev/compute_sanitizer.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jlowe) [reply docs/dev/compute_sanitizer.md:34]: My comment wasn't specific to a single worker mode, it should work for an arbitrary number of workers.  Basically I'm wondering if we can apply the same technique we're using for Nsight profiling standalone clusters to run the sanitizer.  Then we don't have two different approaches for something that needs to wrap the executor process.

[the remark this answers is not in the document]
Comment (rongou) [reply docs/dev/compute_sanitizer.md:34]: I guess it's slightly different from profiling. When profiling you probably only care about single gpu performance, and wrapping the worker and shell is enough. But running the compute sanitizer might benefit from using all the gpus and multiple nodes to shake out the rare bug. Wrapping java would give you more flexibility here.

**Review by jlowe — APPROVED** [review]

**Comment by jlowe** [comment]
build

**Pull request merged**